### ISSUE TITLE
Enhancement: Add class and funcion name to `ImproperlyConfiguredException`s

### ISSUE THREAD

**Issue by rafalkrupinski** [litestar-org/litestar#3548]
### Summary

When you forget to add a type annotation in a route handler function, Litestar raises ImproperlyConfiguredException but neither the stack trace, nor the message shows which function or method is in error.

If the annotation were checked in `@get` decorators, the stack trace would let you find the line with the problem. If that's impossible, the exception should at least show function name, and the class in case of methods.

I volunteer to add function name in `_validate_handler_function()`.

**Comment by euri10** [comment]
I definitely +1 this : I found that when switching from another framework, sometimes you make big refactors, forget some stuff and get hit by that and it's hard to find where you made the mistakes as you just changed too many things, this would definitly make the process smoother.
There are obviouly other cases for that, but this one comes to mind :)

**Comment by rafalkrupinski** [comment]
> when switching from another framework

That's exactly where I'm coming from

**Comment by euri10** [comment]
I should have listed them, but there were other cases where this `ImproperlyConfiguredException` popped up in case of big refactors, and wasn't too useful because of the lack of more granular info. Would that be a good time to list where they're used and maybe investigate if providing more info would make sense ?
Anyways a goo 1st step would be this, thanks for raising it.

**Comment by provinzkraut** [comment]
>  Would that be a good time to list where they're used and maybe investigate if providing more info would make sense ?

Yeah, absolutely. I think we can generally improve a lot of our error messages with more details to make them more useful, but it's going to be a longer process I think. Incremental PRs are definitely welcome there! :)

**Comment by provinzkraut** [comment]
Implemented in #3575.
